File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0 || x != 0 && x % 10 == 0) return false;
        int res = 0; int o = x;
        while(x > res) {
            res = res * 10 + x % 10;
            x /= 10;
        }
        // cout<<res;
        return res == x || res / 10 == x;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        
        if(x<0) {
            return false;
        }
        
        int remainder = 0;
        int oppositeX = 0;
        int temp = x;
            
        while(temp>0){
            remainder = temp % 10;
            oppositeX *= 10;
            oppositeX += remainder;
            temp = temp/10;
        }
        
        return (oppositeX == x);
    }
}